Library/Homebrew/test/spec_helper.rb

```diff
@@ -45,6 +45,20 @@
 
 require_relative "../global"
 
+if HOMEBREW_READELF_RB || HOMEBREW_PATCHELF_RB
```

Comment:
I'm guessing this is related to the problems you had with the unit tests; could you briefly describe why the fixes here are necessary (just for context/retrospection)?

Reply:
left the issue as a comment